ext/standard/basic_functions.c

```diff
@@ -2624,3 +2624,19 @@ PHP_FUNCTION(sys_getloadavg)
 }
 /* }}} */
 #endif
+
+/* {{{ */
+PHP_FUNCTION(nproc)
+{
+	ZEND_PARSE_PARAMETERS_NONE();
+
+	#if defined(_SC_NPROCESSORS_ONLN)
+	int nprocs = sysconf(_SC_NPROCESSORS_ONLN);
+	if (nprocs > 0) {
+		RETURN_LONG(nprocs);
+	}
+	#endif
+
+	RETURN_LONG(1);
```

Comment:
This is not really good way to inform about errors or being unsupported by the platform. It should probably error somehow if there is an error (either warning or exception) and similarly inform user that the function is not support or even not expose it at all if it is not supported.

Reply:
> This is not really good way to inform about errors or being unsupported by the platform.

I understand. Do you think we can keep the default 0 now that windows is also supported? I assume the implementation should work for most OSs.


> either warning or exception

What exception would fit here? 

> even not expose it at all if it is not supported.

Similar to sys_getloadavg. 
We would expose the function for every OS with _SC_NPROCESSORS_ONLN and for windows. 
The disadvantage is that a function_exists call is necessary before using the function.

Reply:
Returning `NULL` might also be an option to allow `cpu_cores() ?? 1` (or maybe implement it yourself for the given architecture). The signature will also make it more obvious that this can fail. A warning is hard to catch and handle (unless the return value is also distinguishable).